### PR TITLE
[MISC] basic editorconfig

### LINKED ISSUE #1383: add an EditorConfig for code formatting
https://editorconfig.org/

### PR DESCRIPTION
**Pull request by mr-c**
Fixes #1383

### CODE REVIEW

**Comment by smehringer** [comment]
Can you add 

```
insert_final_newline = true
```

thanks :)

**Comment by mr-c** [comment]
@smehringer Done!

**Comment by marehr** [comment]
[`seqan3/include/seqan3/std/`](https://github.com/seqan/seqan3/tree/master/include/seqan3/std) has no file extensions, so it would be good to add them explicitly.

**Comment by mr-c** [comment]
@marehr Done, thanks!

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# https://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+[{**.{cpp,hpp},include/seqan3/std/*}]
+indent_style = space
+indent_size = 4
+charser = utf-8
```
Comment (marehr) [inline .editorconfig]:
```suggestion
charset = utf-8
```

**Comment by h-2** [comment]
> @h-2, @rrahn What do you think?

Never heard of it, but also don't mind it.

**Comment by rrahn** [comment]
> @h-2, @rrahn What do you think?

yes why not. If that helps to conform some basic style guides this would be useful already.
Though I believe a note on the tutorials for setting up the editor would be a nice addition to that.

**Comment by mr-c** [comment]
> Though I believe a note on the tutorials for setting up the editor would be a nice addition to that.

CONTRIBUTING.md links to https://github.com/seqan/seqan3/wiki/General-formatting but I don't have permission to edit that. If I did, I would add a link to https://editorconfig.org/#download

**Comment by smehringer** [comment]
> CONTRIBUTING.md links to https://github.com/seqan/seqan3/wiki/General-formatting but I don't have permission to edit that. If I did, I would add a link to https://editorconfig.org/#download

@eseiler will take care of it

**Pull request merged**